File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel

  def subscribed
    stream_from room_name
  end

  def say(data)
    ActionCable.server.broadcast(room_name, {
      message: data['message']
    })
  end

  private
  def room_name
    'cable:global_chat_room'
  end
end

File: app/assets/javascripts/channels/chat_channel.js
var chatChannel = App.cable.subscriptions.create("ChatChannel", {
  connected: function () {
  },

  received: function (data) {
    console.log("Message: " + data.message)
  },

  say: function (message) {
    this.perform("say", {message: message})
  }
})
